src/agentConnection.js

```diff
@@ -241,3 +244,98 @@ function sendData(path, data, cb, ignore404) {
   req.write(payload);
   req.end();
 }
+
+/**
+ * Sends the given event data and trace data synchronously to the agent via HTTP. This function is synchronous, that is,
+ * it blocks the event loop!
+ *
+ * YOU MUST NOT USE THIS FUNCTION, except for the one use case where it is actually required to block the event loop
+ * (reporting an uncaught exception tot the agent in the process.on('uncaughtException') handler).
+ */
+exports.reportUncaughtExceptionToAgentSync = function reportUncaughtExceptionToAgentSync(eventData, spans) {
+  sendRequestsSync([{
+    path: '/com.instana.plugin.nodejs/traces.' + pidStore.pid,
+    data: spans
+  }, {
+    path: '/com.instana.plugin.generic.event',
+    data: eventData
+  }]);
+};
+
+
+/**
+ * Sends multiple HTTP POST requests to the agent. This function is synchronous, that is, it blocks the event loop!
+ *
+ * YOU MUST NOT USE THIS FUNCTION, except for the one use case where it is actually required to block the event loop
+ * (reporting an uncaught exception tot the agent in the process.on('uncaughtException') handler).
+ */
+function sendRequestsSync(requests) {
+  // only try to require optional dependency netlinkwrapper once
+  if (!netLinkHasBeenRequired) {
+    netLinkHasBeenRequired = true;
+    try {
+      netLink = require('netlinkwrapper')();
+    } catch (requireNetlinkError) {
+      logger.warn('Failed to require optional dependency netlinkwrapper, uncaught exception will not be reported ' +
+                  'to Instana.');
+    }
+  }
+  if (!netLink) {
+    return;
+  }
+
+  var port = agentOpts.port;
+  if (typeof port !== 'number') {
+    try {
+      port = parseInt(port, 10);
+    } catch (nonNumericPortError) {
+      logger.warn('Detected non-numeric port configuration value %s, uncaught exception will not be reported.', port);
+      return;
+    }
+  }
+
+  try {
+    netLink.connect(port, agentOpts.host);
+    netLink.blocking(false);
```

Comment:
Shouldn't this be `true`? (I didn't read the docs)

Reply:
No, you don't won't the socket to be in blocking mode - this has nothing to do with the call blocking the event loop, it's a OS socket level thing (see https://www.scottklement.com/rpg/socktut/nonblocking.html for an explanation). We want to write to the socket and then end the process. If we can't write, we want to simply fail early instead of blocking indefinitely.